libs/payments/ui/src/lib/client/components/PaymentMethodManagement/index.tsx

```diff
@@ -0,0 +1,228 @@
+// /* This Source Code Form is subject to the terms of the Mozilla Public
+// * License, v. 2.0. If a copy of the MPL was not distributed with this
+// * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+'use client';
+
+import {
+  PaymentElement,
+  useElements,
+  useStripe,
+} from '@stripe/react-stripe-js';
+import * as Form from '@radix-ui/react-form';
+import { LoadingSpinner } from '../LoadingSpinner';
+import { useState, useRef } from 'react';
+import { StripePaymentElementChangeEvent } from '@stripe/stripe-js';
+import { Localized, useLocalization } from '@fluent/react';
+import { useRouter } from 'next/navigation';
+
+export function PaymentMethodManagement({
+  onSubmit,
+  defaultPaymentMethodId,
+}: {
+  onSubmit: (confirmationTokenId: string) => any;
+  defaultPaymentMethodId: string;
+}) {
+  const { l10n } = useLocalization();
+  const stripe = useStripe();
+  const elements = useElements();
+  const formRef = useRef<HTMLFormElement>(null);
+  const router = useRouter();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isComplete, setIsComplete] = useState(false);
+  const [isReady, setIsReady] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [isInputNewCardDetails, setIsInputNewCardDetails] = useState(false);
+  const [fullName, setFullName] = useState('');
+  const [hasFullNameError, setHasFullNameError] = useState(false);
+  const [isNonDefaultCardSelected, setIsNonDefaultCardSelected] =
+    useState(false);
+
+  const handleReady = () => {
+    setIsReady(true);
+  };
+
+  const handlePaymentElementChange = (
+    event: StripePaymentElementChangeEvent
+  ) => {
+    console.log('event', event);
+    setIsComplete(event.complete);
+
+    if (event.value.type === 'card' && !event.value.payment_method) {
+      setIsInputNewCardDetails(true);
+
+      if (event.complete) {
+        setHasFullNameError(fullName.length === 0);
+      }
+    } else if (event.value.type === 'card' && !!event.value.payment_method) {
+      setIsInputNewCardDetails(false);
+
+      if (event.value.payment_method.id !== defaultPaymentMethodId) {
+        setIsNonDefaultCardSelected(true);
+      } else {
+        setIsNonDefaultCardSelected(false);
+      }
+    }
+  };
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!stripe || !elements || !isComplete) {
+      return;
+    }
+
+    setIsLoading(true);
+    setError(null);
+
+    try {
+      elements.submit();
```

Comment:
issue: `elements.submit()` returns a Promise and needs to be awaited

Please see [the Stripe docs](https://docs.stripe.com/js/elements/submit) for more information, or have a look at `CheckoutForm/index.ts`